keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -157,21 +159,17 @@ public DoubleTensor times(double that) {
 
     @Override
     public DoubleTensor matrixMultiply(DoubleTensor that) {
-        if (that.isLengthOne()) {
+        if (shape.length == 2) {
             return that.times(value);
+        } else {
+            throw new IllegalArgumentException("Cannot matrix multiply shape: " + Arrays.toString(shape));
```

Comment:
you can use `MatrixMultiplicationVertex#getResultingShape` - move it to `TensorShapeValidation`

Reply:
👍